components/MaskedTransition/src/private/MDCMaskedTransition.m

```diff
@@ -46,83 +46,118 @@ static inline CGFloat LengthOfVector(CGVector vector) {
   return (CGFloat)hypot(vector.dx, vector.dy);
 }
 
-@interface MDCMaskedTransition () <MDMTransitionWithPresentation, MDMTransitionWithFeasibility>
-@end
+// TODO: Pull this out to MotionTransitioning.
+static NSArray <UIViewController *> *
+PrepareTransitionWithContext(id<UIViewControllerContextTransitioning> transitionContext,
+                             MDMTransitionDirection direction) {
+  UIViewController *from =
+      [transitionContext viewControllerForKey:UITransitionContextFromViewControllerKey];
+  UIView *fromView = [transitionContext viewForKey:UITransitionContextFromViewKey];
+  if (fromView == nil) {
+    fromView = from.view;
+  }
+  if (fromView != nil && fromView == from.view) {
+    CGRect finalFrame = [transitionContext finalFrameForViewController:from];
+    if (!CGRectIsEmpty(finalFrame)) {
+      fromView.frame = finalFrame;
+    }
+  }
+
+  UIViewController *to =
+      [transitionContext viewControllerForKey:UITransitionContextToViewControllerKey];
+  UIView *toView = [transitionContext viewForKey:UITransitionContextToViewKey];
+  if (toView == nil) {
+    toView = to.view;
+  }
+  if (toView != nil && toView == to.view) {
+    CGRect finalFrame = [transitionContext finalFrameForViewController:to];
+    if (!CGRectIsEmpty(finalFrame)) {
+      toView.frame = finalFrame;
+    }
+
+    if (toView.superview == nil) {
+      switch (direction) {
+        case MDMTransitionDirectionForward:
+          [transitionContext.containerView addSubview:toView];
+          break;
+
+        case MDMTransitionDirectionBackward:
+          [transitionContext.containerView insertSubview:toView atIndex:0];
+          break;
+      }
+    }
+  }
+
+  [toView layoutIfNeeded];
+
+  if (direction == MDMTransitionDirectionForward) {
+    return @[from, to];
+  } else {
+    return @[to, from];
+  }
+}
 
 @implementation MDCMaskedTransition {
   UIView *_sourceView;
-  BOOL _shouldSlideWhenCollapsed;
+  MDMTransitionDirection _direction;
 }
 
-- (instancetype)initWithSourceView:(UIView *)sourceView {
+- (instancetype)initWithSourceView:(UIView *)sourceView
+                         direction:(MDMTransitionDirection)direction {
   self = [super init];
   if (self) {
     _sourceView = sourceView;
+    _direction = direction;
   }
   return self;
 }
 
-- (BOOL)canPerformTransitionWithContext:(__unused id<MDMTransitionContext>)context {
-  return _shouldSlideWhenCollapsed ? NO : YES;
-}
-
-#pragma mark - MDMTransitionWithPresentation
+#pragma mark - UIViewControllerAnimatedTransitioning
 
-- (UIModalPresentationStyle)defaultModalPresentationStyle {
-  return UIModalPresentationCustom;
+- (NSTimeInterval)transitionDuration:(id<UIViewControllerContextTransitioning>)transitionContext {
+  return 0.375;
```

Comment:
Is this timing documented in a specification somewhere?

Reply:
Added a spec parameter.